lib/rom/repository/transaction.rb

```diff
@@ -0,0 +1,41 @@
+module ROM
+  class Repository
+    class Transaction
+      attr_reader :repo
+
+      attr_reader :ops
+
+      def initialize(repo, ops = [])
+        @repo = repo
+        @ops = ops
+      end
+
+      def create(changeset)
+        new(ops.dup << [:create, changeset])
```

Comment:
Yeah, I actually started with a transaction that accumulates its ops, then changed it. Not sure if that was a good idea. I need to experiment with more use cases and we'll see what makes more sense.